PYIC-6685: Use `userinfo` claim, not `userInfo`

Wynndow opened the pull request:
<!-- Provide a general summary of your changes in the Title above -->
<!-- Include the Jira ticket number in square brackets as prefix, eg `[P4-XXXX] PR Title` -->

## Proposed changes

### What changed

Use `userinfo` claim, not `userInfo`

### Why did it change

We were extracting the userinfo claim using the wrong claim name. Our orch-stub was sending us the wrong claim name, so this was only detected in staging when running a journey using the real orch.


### Issue tracking
<!-- List any related Jira tickets or GitHub issues -->
<!-- List any related ADRs or RFCs -->
<!-- Delete/copy as appropriate -->

- [PYIC-6685](https://govukverify.atlassian.net/browse/PYIC-6685)


[PYIC-6685]: https://govukverify.atlassian.net/browse/PYIC-6685?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

Wynndow committed:
PYIC-6685: Use userinfo claim, not userInfo

We were extracting the userinfo claim using the wrong claim name. Our
orch-stub was sending us the wrong claim name, so this was only detected
in staging when running a journey using the real orch.

Joe-Edwards-GDS reviewed (dismissed):
Maybe we should prioritise that checking the schema ticket!

rpayal committed:
PYIC-6685 Refactor code and updated test after 'userinfo' claim, not 'userInfo' under claims in jar frm Orch.

Joe-Edwards-GDS commented on lambdas/initialise-ipv-session/src/main/java/uk/gov/di/ipv/core/initialiseipvsession/InitialiseIpvSessionHandler.java:
```diff
@@ -193,10 +193,11 @@ public APIGatewayProxyResponseEvent handleRequest(
-                    || configService.enabled(EVCS_WRITE_ENABLED))) {
-                Optional<JarUserInfo> jarUserInfoClaim = getJarUserInfo(claimsSet);
+                            || configService.enabled(EVCS_WRITE_ENABLED))
+                    && (jarUserInfoClaim.isPresent())) {
```
Do you need this additional check? If anything I think it's problematic because it means we'll silently ignore a missing access token if we need it.

rpayal replied:
yeah don't need this check.....as it'll already be taken care in validation.

Wynndow committed:
PYIC-6684: Move userinfo parsing behind flag

This means we're only trying to parse the claim when needed. We're also
allowing the optional to do its job.
